cnc_ctrl_v1/libraries/Axis/axis.cpp

```diff
@@ -236,43 +236,160 @@ void   Axis::printBoost(){
     _disableAxisForTesting = false;
 }
 
-void   Axis::measureMotorSpeed(int speed){
+void   Axis::computeMotorResponse(){
+    Serial.println("Compute motor response");
+    
+    //remove whatever transform is applied
+    _motor.setSegment(0 , 1, 0, 0, 0);
+    _motor.setSegment(1 , 1, 0, 0, 0);
+    _motor.setSegment(2 , 1, 0, 0, 0);
+    _motor.setSegment(3 , 1, 0, 0, 0);
+    
+    //In the positive direction
+    //-----------------------------------------------------------------------------------
+    
+    float scale = 255/measureMotorSpeed(255); //Y3*scale = 255 -> scale = 255/Y3
+    
+    int i = 15;
+    float motorSpeed;                                                                                                                                     
+    while (i < 255){
+        motorSpeed = measureMotorSpeed(i);
+        
+        if (motorSpeed > 0){
+            break;
+        }
+        
+        i++;
+    }
+    
+    float X1 = i;
+    float Y1 = scale*motorSpeed;
+    
+    float X2 = (255 - X1)/2;
+    float Y2 = scale*measureMotorSpeed(X2);
+    
+    float X3 = 255;
+    float Y3 = 255;
+    
+    float M1 = (Y1 - Y2)/(X1 - X2);
+    float M2 = (Y2 - Y3)/(X2 - X3);
+    
+    float I1 = -1*(Y1 - (M1*X1));
+    float I2 = -1*(Y2 - (M2*X2));
+    
+    _motor.setSegment(2 , M1, I1,    0,   Y2);
+    _motor.setSegment(3 , M2, I2, Y2-1, Y3+1);
+    
+    Serial.println("pos done");
+    //In the negative direction
+    //-----------------------------------------------------------------------------
+    
+    scale = -255/measureMotorSpeed(-255); //Y3*scale = 255 -> scale = 255/Y3
+    
+    i = -30;                                                                                                                                  
+    while (i > -255){
+        motorSpeed = measureMotorSpeed(i);
+        
+        if (motorSpeed < 0){
+            break;
+        }
+        
+        i--;
+    }
+    
+    X1 = i;
+    Y1 = scale*motorSpeed;
+    
+    X2 = (-255 - X1)/3 + X1;
+    Y2 = scale*measureMotorSpeed(X2);
+    
+    X3 = -255;
```

Comment:
going to leave this for the next branch because I want to test after making that change